File: Domain/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Model;

namespace Domain
{
    public class Order : BaseDeletableModel<Guid>
    {
        public Order()
        {
            this.Id = Guid.NewGuid();
        }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int OrderNumber { get; set; }

        public Guid PartnerId { get; set; }

        public virtual Partner Partner { get; set; }

        public Guid WarehouseId { get; set; }

        public virtual Warehouse Warehouse { get; set; }

        public bool IsCompleted { get; set; }
        public Guid DeliveryAddressId { get; set; }
        public virtual DeliveryAddress DeliveryAddress { get; set; }

        public DateTime? CompletedDate { get; set; }

        public string CompletedBy { get; set; }

        [ForeignKey("CompletedBy")]
        public virtual AppUser CompletedByUser { get; set; }

        public string CreatedBy { get; set; }

        [ForeignKey("CreatedBy")]
        public virtual AppUser CreatedByUser { get; set; }

        public virtual ICollection<OrderProduct> OrderProducts { get; set; }
    }
}

File: client-app/src/app/models/order.ts
import { OrderProduct } from "./orderProduct";

export interface Order {
    id: string;
    orderNumber: number;
    isCompleted: boolean;
    completedDate: Date;
    orderCreated: Date;
    partnerId: string;
    partnerName: string;
    city: string;
    companyOwnerName: string;
    email: string;
    phone: string;
    address: string;
    bulstat: string;
    deliveryCity: string;
    deliveryAddress: string;
    warehouseId: string;
    warehouseName: string;
    contactPersonId: string;
    contactPersonName: string;
    contactPersonPhoneNumber: string;
    orderProductDtos: Array<OrderProduct>;
    completedBy: string;
    createdBy: string;
}

export interface OrderFormValues {
    id: string;
    partnerId: string;
    deliveryAddressId: string;
    warehouseId: string;
    createdBy: string;
    orderProducts: Array<OrderProduct>;
}
